src/terminal/frame.rs

```diff
@@ -89,10 +89,10 @@ impl Frame<'_> {
     /// # let mut frame = terminal.get_frame();
     /// let block = Block::default();
     /// let area = Rect::new(0, 0, 5, 5);
-    /// frame.render_widget_ref(block, area);
+    /// frame.render_widget_ref(&block, area);
     /// ```
     #[stability::unstable(feature = "widget-ref")]
-    pub fn render_widget_ref<W: WidgetRef>(&mut self, widget: W, area: Rect) {
+    pub fn render_widget_ref<W: WidgetRef>(&mut self, widget: &W, area: Rect) {
```

Comment:
W can also contain `&WidgetRef` or `&mut WidgetRef` :thinking: 

https://github.com/pretzelhammer/rust-blog/blob/master/posts/common-rust-lifetime-misconceptions.md/#1-t-only-contains-owned-types

Reply:
```rust
impl Widget for &Something {…}
```

This means that a reference of something is enough to render… no need to transfer the ownership… I think this is a bigger topic that should be moved out of this PR?

Reply:
The main reason this is unstable is because there's some possible options for implementing this (noted at https://users.rust-lang.org/t/calling-a-method-with-a-self-parameter-on-box-dyn-trait/106220) that would be slightly incompatible.

The history on this is that Widget takes `self` instead of &self, and changing that breaks everything, so we needed something more interesting to make &self and Box<dyn Widget> to work. It's possible that we didn't actually need something more interesting in the long run. It's also possible that we could just break everything instead.

The relevant PRs for more context are:
- https://github.com/ratatui-org/ratatui/pull/833
- https://github.com/ratatui-org/ratatui/pull/903
- https://github.com/ratatui-org/ratatui/pull/897

Reply:
I allowed this lint on this. It should be discussed in its own place. When searching for allowed lints this will come up again.